addons/common/functions/fnc_registerItemReplacement.sqf

```diff
@@ -0,0 +1,114 @@
+#include "script_component.hpp"
+/*
+ * Author: BaerMitUmlaut
+ * Registers an event handler that replaces an item or item type with one or
+ * more other items.
+ *
+ * Arguments:
+ * 0: Item or item type ID to replace <STRING or NUMBER>
+ * 1: Item or list of items <STRING or ARRAY>
+ * 2: Replace items that inherit from original item (only if 0 is STRING) <BOOL>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * ["FirstAidKit", "ACE_fieldDressing"] call ace_common_fnc_registerItemReplacement
+ *
+ * Public: Yes
+ */
+params ["_oldItem", "_newItems", ["_replaceInherited", false]];
+
+// CBA player event handler function
+private _fnc_replaceItems = {
+    params ["_unit"];
+
+    private _items = items _unit;
+    if (_items isEqualTo GVAR(oldItems)) exitWith {};
+
+    private _newItems = _items - GVAR(oldItems);
+    if (_newItems isEqualTo []) exitWith {
+        GVAR(oldItems) = _items;
+    };
+
+    _newItems sort true; // Sort so all items of current class can be replaced at once
+    private _replacements = [];
+    private _cfgWeapons = configFile >> "CfgWeapons"; // Microoptimization
+
+    for "_i" from 0 to count _newItems - 1 do {
+        private _item = _newItems#_i;
+
+        // Determine replacement items: direct replacements, ...
+        private _directReplacement = GVAR(itemReplacements) getVariable [_item, []];
+        _replacements append _directReplacement;
+
+        // ... item type replacements ...
+        private _type = getNumber (_cfgWeapons >> _item >> "ItemInfo" >> "type");
+        private _typeReplacements = GVAR(itemReplacements) getVariable ["$" + str _type, []];
+        _replacements append _typeReplacements;
+
+        // ... and inherited replacements
+        {
+            if (_item isKindOf [_x, _cfgWeapons]) then {
+                private _inheritedReplacements = GVAR(itemReplacements) getVariable ["$" + _x, []];
+                _replacements append _inheritedReplacements;
+            };
+        } forEach GVAR(inheritedReplacements);
+
+        // Skip lookup for all following items of this class
+        private _count = 1;
+        while {_newItems#(_i + 1) == _item} do {
```

Comment:
This can go out of bounds for the array, but `while {nil}` fails silently so it's probably fine, maybe leave a note